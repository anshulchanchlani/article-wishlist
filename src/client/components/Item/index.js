import React, { Component } from 'react';
import StarRatings from 'react-star-ratings'
import {connect} from 'react-redux'
import {addItemToWishlist,deleteFromWishlist} from '../../actions/wishlistActions'
import './index.scss'
class Item extends Component {
    constructor(props){
        super(props)
        this.state ={isFav:this.props.isFav?this.props.isFav:false}
        this.handleFavClick = this.handleFavClick.bind(this)
    }
    changedImageSize(imgUrl) {
        if (imgUrl) {
            return `${imgUrl.split('?')[0]}?sw=300&sh=300&sm=fit`
        }
    }
    handleFavClick(){
        let item = this.props.item;
        this.setState({isFav:!this.state.isFav},()=>{
            this.state.isFav?this.props.addItemToWishlist(item):this.props.deleteFromWishlist(item);
        })
        
    }
    returnStarRatings(stars){
      let rating = Number.isInteger(+stars)?+stars:0;
      return  <StarRatings
          rating={rating}
          starRatedColor="black"
          changeRating={this.changeRating}
          numberOfStars={5}
          starDimension="20px"
          starSpacing="7px"
          name='rating'
        />
    }
    render() {
        const { item } = this.props
        return (

            <li> 
                <div className="item">
                  
                    <i onClick={this.handleFavClick} className="material-icons add-to-wishlist-icon">
                        {this.state.isFav?'favorite':'favorite_border'}
                    </i>
                 
                    <a href={item.url} target="_blank">
                    <div className="wrapper-style">
                        <div>
                            <img src={this.changedImageSize(item.image)} width="250px" alt={item.subtitle} height="250px" />
                        </div>
                        <div><span>{item.suggestion}</span></div>
                        <div><span className="item-subtitle">{item.subTitle}</span></div>
                        <div><span title="price">{item.salePrice ? item.salePrice : item.standardPrice}</span></div>
                        <div><span title="preorder">{item.isPreorder ? 'Available for Preorder' : 'Not available for preorder.'}</span></div>
                        <div><strong>{this.returnStarRatings(item.rating?item.rating:0)}</strong></div>
                    </div>
                    </a>
                </div>
            </li>

        );
    }
}


export default connect(null,{deleteFromWishlist,addItemToWishlist})(Item);
