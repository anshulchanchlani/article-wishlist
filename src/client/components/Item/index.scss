.item{
    font-size:14px;
    .wrapper-style{
        &:hover{
            border:1px solid rgb(199, 198, 198);
        }
    }
}

.item-subtitle{
    color:rgb(168, 164, 164);
    text-decoration: none;
}
.add-to-wishlist-icon{
    position: relative;
    top: 35px;
    left: 107px;
    cursor:pointer
}
