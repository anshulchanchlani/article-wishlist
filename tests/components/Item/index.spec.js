import React from 'react';
import {mockStore} from '../../setup/mockStore'
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import {mount,shallow} from 'enzyme';
import toJson from 'enzyme-to-json'
import Item from '../../../src/client/components/Item'


const store = mockStore({wishlist:[]})
const product = {
    suggestion:'abc',
    image:'abc',
    subTitle:'abc',
    salePrice:'38',
    isPreorder:"",
    rating:4,
    reviews:23
}
describe('Item renders',()=>{
    let isFavItem = jest.fn();
    const component = <Provider store={store}><Item isFav={isFavItem(product)} item={product}/></Provider>
    it('Item Component renders correctly',()=>{
        const wrapper = shallow(component)
        const jsonWrapper = toJson(wrapper);
        const testInstance = mount(component);
        
        expect(jsonWrapper).toMatchSnapshot();

    })
   
})