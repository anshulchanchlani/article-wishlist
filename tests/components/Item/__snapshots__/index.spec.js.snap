// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Item renders Item Component renders correctly 1`] = `
<ContextProvider
  value={
    Object {
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "storeState": Object {
        "wishlist": Array [],
      },
    }
  }
>
  <Connect(Item)
    item={
      Object {
        "image": "abc",
        "isPreorder": "",
        "rating": 4,
        "reviews": 23,
        "salePrice": "38",
        "subTitle": "abc",
        "suggestion": "abc",
      }
    }
  />
</ContextProvider>
`;
